### PR TITLE
Adding remember me option for Business Rules Manager

### PR DESCRIPTION
**Pull request by senthuran16**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: components/org.wso2.carbon.business.rules.web/src/utils/AuthManager.jsx [hunk 1]
```diff
@@ -101,9 +140,11 @@ export default class AuthManager {
                 .logout(AuthManager.getUser().token)
                 .then(() => {
                     AuthManager.clearUser();
+                    window.localStorage.clear();
+                    AuthManager.delete_cookie("REFRESH_TOKEN");
```
Comment (lasanthaS) [inline components/org.wso2.carbon.business.rules.web/src/utils/AuthManager.jsx]: Shall we rename this to `deleteCookie`?

File: components/org.wso2.carbon.business.rules.web/src/api/AuthenticationAPI.jsx [hunk 2]
```diff
@@ -54,6 +55,26 @@ export default class AuthenticationAPI {
...
+        return AuthenticationAPI
+            .getHttpClient()
+            .post(`/login/${appContext}`, Qs.stringify({
+                grantType: "refresh_token",
+                rememberMe: true
+            }), {
+                headers: {
+                    'Content-Type': MediaType.APPLICATION_WWW_FORM_URLENCODED,
+                    'Authorization': "Bearer " + AuthManager.getCookie('REFRESH_TOKEN'),
+                    'Accept': 'application/json'
```
Comment (lasanthaS) [inline components/org.wso2.carbon.business.rules.web/src/api/AuthenticationAPI.jsx]: Let's move this to `MediaType` constants as well.

File: components/org.wso2.carbon.business.rules.web/src/components/auth/Login.jsx [hunk 3]
```diff
@@ -94,7 +103,7 @@ export default class Login extends React.Component {
             .then(() => this.setState({authenticated: true}))
             .catch((error) => {
                 const errorMessage = error.response && error.response.status === 401 ?
-                    'The username/password is invalid' : 'Unknown error occurred!';
+                    'The username/password is invalid' : 'Unknown error occurred!';;
```
Comment (niruhan) [inline components/org.wso2.carbon.business.rules.web/src/components/auth/Login.jsx:106]: Is it possible to check for other HTTP codes and provide error messages?

File: components/org.wso2.carbon.business.rules.web/src/utils/AuthManager.jsx [hunk 4]
```diff
@@ -153,4 +194,51 @@ export default class AuthManager {
...
+        if (validityPeriod) {
+            const date = new Date();
+            date.setTime(date.getTime() + validityPeriod * 1000);
+            expires = "; expires=" + date.toUTCString();
+        }
+        document.cookie = name + "=" + value + expires + "; path=" + path + securedDirective;
+    }
+
+    /**
+     * Get JavaScript accessible cookies saved in browser, by giving the cooke name.
```
Comment (niruhan) [inline components/org.wso2.carbon.business.rules.web/src/utils/AuthManager.jsx:218]: small typo "cooke" -> "cookie"

File: components/org.wso2.carbon.business.rules.web/src/utils/AuthManager.jsx [hunk 5]
```diff
@@ -153,4 +194,51 @@ export default class AuthManager {
...
+     * @param {String} name : Name of the cookie which need to be retrived
+     * @returns {String|null} : If found a cookie with given name , return its value,Else null value is returned
```
Comment (niruhan) [inline components/org.wso2.carbon.business.rules.web/src/utils/AuthManager.jsx:220]: shall we add a space after "value,"

File: components/org.wso2.carbon.business.rules.web/src/utils/AuthManager.jsx [hunk 6]
```diff
@@ -153,4 +194,51 @@ export default class AuthManager {
...
+     */
+    static getCookie(name) {
+        let pairs = document.cookie.split(";");
+        let cookie = null;
+        for (let pair of pairs) {
+            pair = pair.split("=");
+            let cookie_name = pair[0].trim();
```
Comment (niruhan) [inline components/org.wso2.carbon.business.rules.web/src/utils/AuthManager.jsx:227]: shall we continue using camelcase?

File: components/org.wso2.carbon.business.rules.web/src/components/auth/Login.jsx (hunk 3 above)
Comment (senthuran16) [reply components/org.wso2.carbon.business.rules.web/src/components/auth/Login.jsx:106]: Have followed the same approach as https://github.com/wso2/carbon-dashboards/blob/master/components/dashboards-web-component/src/auth/Login.jsx#L188.

File: components/org.wso2.carbon.business.rules.web/src/utils/AuthManager.jsx (hunk 6 above)
Comment (senthuran16) [reply components/org.wso2.carbon.business.rules.web/src/utils/AuthManager.jsx:227]: This function (`getCookie(name)`) doesn't contain `cookie_name` anymore. _(See https://github.com/wso2/carbon-analytics/blob/master/components/org.wso2.carbon.business.rules.web/src/utils/AuthManager.js#L303)_

File: components/org.wso2.carbon.business.rules.web/src/utils/AuthManager.jsx (hunk 4 above)
Comment (senthuran16) [reply components/org.wso2.carbon.business.rules.web/src/utils/AuthManager.jsx:218]: Has been already fixed with https://github.com/wso2/carbon-analytics/commit/899c8383f8deb99f6449e32e717a623d717152bb#diff-d6451e6f6300529a450ad777b4828b2aR223. _(Currently it looks like [this](https://github.com/wso2/carbon-analytics/blame/master/components/org.wso2.carbon.business.rules.web/src/utils/AuthManager.js#L299))_.

File: components/org.wso2.carbon.business.rules.web/src/utils/AuthManager.jsx (hunk 5 above)
Comment (senthuran16) [reply components/org.wso2.carbon.business.rules.web/src/utils/AuthManager.jsx:220]: Has been already fixed with https://github.com/wso2/carbon-analytics/commit/899c8383f8deb99f6449e32e717a623d717152bb#diff-d6451e6f6300529a450ad777b4828b2aR225. _(Currently it looks like [this](https://github.com/wso2/carbon-analytics/blame/master/components/org.wso2.carbon.business.rules.web/src/utils/AuthManager.js#L301))_.

**Pull request merged**